### PR TITLE
feat(ST-769): implement the localization supporting

### PR DESCRIPTION
**Pull request by trueboroda**
Description:

--

QA-test: ST-1059

Demo-test: ST-1060

Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/client-app-0.0.4-pr-85-32e0cfe8.zip

### CODE REVIEW

File: client-app/pages/account/addresses.vue [hunk 1]
```diff
@@ -13,7 +13,11 @@
             <h2 class="text-gray-800 text-3xl font-bold uppercase">{{ title }}</h2>
 
             <VcButton v-if="!editingMode" class="px-3 uppercase border" size="sm" is-outline @click="openEditMode()">
-              {{ isMobile ? "Add new" : "Add new address" }}
+              {{
+                isMobile
+                  ? $t("pages.account.addresses.add_new_address_mobile_button")
```
Comment (Flanker72) [inline client-app/pages/account/addresses.vue:18]: Maybe better to use modificator as suffix: `add_new_address_button_mobile`

File: client-app/pages/account/order-details.vue [hunk 2]
```diff
@@ -6,19 +6,26 @@
 
     <div class="max-w-screen-2xl md:px-12 mx-auto">
       <h2 class="text-gray-800 px-5 md:px-0 text-2xl lg:text-3xl font-bold uppercase mb-2">
-        Order #{{ order?.number }}
+        {{ $t("pages.account.order_details.title_prefix") }}{{ order?.number }}
```
Comment (Flanker72) [inline client-app/pages/account/order-details.vue:9]: Need to use formatted localized string. See https://vue-i18n.intlify.dev/guide/essentials/syntax.html as an example.

File: client-app/pages/bulk-order/bulk-order.vue [hunk 3]
```diff
@@ -67,10 +70,13 @@ import { CopyAndPaste, Manually } from "@/shared/bulk-order";
...
 import { useRouter } from "vue-router";
+import { useI18n } from "vue-i18n";
+
+const { t } = useI18n();
 
 const tabs = [
-  { id: "manually", label: "Manually" },
-  { id: "copy&paste", label: "Copy & Paste" },
+  { id: "manually", label: t("pages.bulk_order.manually_tab") },
+  { id: "copy&paste", label: t("pages.bulk_order.copy_n_past_tab") },
```
Comment (Flanker72) [inline client-app/pages/bulk-order/bulk-order.vue:79]: Missing "e" in "paste" word.

File: client-app/pages/product/product.vue [hunk 4]
```diff
@@ -36,18 +36,18 @@
...
           <!-- Price & Delivery (without variations) -->
           <ProductPriceBlock v-else :product="product">
             <div class="flex items-baseline justify-between text-sm">
-              <div class="font-extrabold text-base">Your price:</div>
+              <div class="font-extrabold text-base" v-t="'pages.product.price_label'"></div>
 
               <div>
                 <VcPriceDisplay :value="product.price?.actual" class="font-extrabold text-green-700" />
-                <span class="font-semibold hidden lg:inline-block ml-1">/ each</span>
+                <span class="font-semibold hidden lg:inline-block ml-1" v-t="'common.suffixes.product_price'"></span>
```
Comment (Flanker72) [inline client-app/pages/product/product.vue:50]: Incorrect key name. "Product price" suffix is usually understanded as currency or measure unit, etc. This key should be renamed to "per_item".

File: client-app/pages/reset-password/reset-password.vue [hunk 5]
```diff
@@ -1,27 +1,30 @@
 <template>
   <TwoColumn class="max-w-screen-xl">
     <template #left>
-      <h1 class="uppercase tracking-wide text-3xl lg:text-4xl font-bold mb-8 lg:mb-9 lg:mt-6">reset your password</h1>
+      <h1
+        class="uppercase tracking-wide text-3xl lg:text-4xl font-bold mb-8 lg:mb-9 lg:mt-6"
+        v-t="'pages.reset-password.header'"
```
Comment (Flanker72) [inline client-app/pages/reset-password/reset-password.vue:6]: Why "reset-password" is dashed? All previous similar page keys were renamed to underscore notation.

File: client-app/pages/sign-up/sign-up-page.vue [hunk 6]
```diff
@@ -1,56 +1,56 @@
 <template>
   <TwoColumn class="max-w-screen-xl">
     <template #left>
-      <h1 class="uppercase tracking-wide text-3xl lg:text-4xl font-bold mb-8 lg:mt-5">Registration</h1>
+      <h1 class="uppercase tracking-wide text-3xl lg:text-4xl font-bold mb-8 lg:mt-5" v-t="'pages.sign-up.header'"></h1>
```
Comment (Flanker72) [inline client-app/pages/sign-up/sign-up-page.vue:4]: Why "sign-up" is dashed? All previous similar page keys were renamed to underscore notation.

File: client-app/pages/account/addresses.vue (hunk 1 above)
Comment (trueboroda) [reply client-app/pages/account/addresses.vue:18]: maybe mobile_add_new_address_button

File: client-app/shared/account/components/product-card-reorder.vue [hunk 7]
```diff
@@ -19,25 +19,35 @@
...
-            <span class="text-xs text-gray-400" v-else> Item is out of stock </span>
+            <span
+              class="text-xs text-gray-400"
+              v-if="!isOutOfStock"
+              v-t="$t('shared.account.reorder_info_popup.product_card.item_can_t_be_purchased_message')"
+            ></span>
+            <span
+              class="text-xs text-gray-400"
+              v-else
+              v-t="$t('shared.account.reorder_info_popup.product_card.item_can_t_be_purchased_message')"
```
Comment (Flanker72) [inline client-app/shared/account/components/product-card-reorder.vue:43]: Wrong message key. Here it should be `item_out_of_stock_message`.

File: client-app/shared/account/components/product-card-reorder.vue [hunk 8]
```diff
@@ -19,25 +19,35 @@
...
+            ></span>
           </div>
         </div>
       </div>
       <div class="flex justify-between">
         <div class="flex items-center space-x-3">
-          <span class="self-start">Quantity:</span>
+          <span class="self-start">{{ $t("shared.account.reorder_info_popup.product_card.quantity_label") }}</span>
```
Comment (Flanker72) [inline client-app/shared/account/components/product-card-reorder.vue:50]: Why this key is in inside `reorder_info_popup` namespace?

File: client-app/shared/account/components/product-card-reorder.vue (hunk 7 above)
Comment (Flanker72) [inline client-app/shared/account/components/product-card-reorder.vue:43]: Wrong key, it should be `item_out_of_stock_message`.

File: client-app/shared/account/components/product-card-reorder.vue [hunk 9]
```diff
@@ -153,17 +176,22 @@
...
                   }}
-                  in stock</span
+                  {{ $t("shared.account.reorder_info_popup.product_card.in_stock_suffix") }}</span
                 >
               </div>
               <div v-else-if="isOutOfStock" class="flex items-center">
-                <span class="text-[color:var(--color-danger)] text-xs pt-1 whitespace-nowrap">Out of stock</span>
+                <span
+                  class="text-[color:var(--color-danger)] text-xs pt-1 whitespace-nowrap"
+                  v-t="'shared.account.reorder_info_popup.product_card.out_of_stock_suffix'"
```
Comment (Flanker72) [inline client-app/shared/account/components/product-card-reorder.vue:185]: Why key contains `suffix` if it is the whole phrase?

File: client-app/shared/account/components/product-card-reorder.vue [hunk 10]
```diff
@@ -19,25 +19,35 @@
...
-              The number of items in the order was
-              <span class="text-gray-500"
-                >reduced from {{ props.productItem.quantity }} to
-                {{ props.productItem.availabilityData?.availableQuantity }}</span
-              >
-              due to stock restrictions
+            <span
+              class="text-xs text-gray-400"
+              v-html="
+                $t('shared.account.reorder_info_popup.product_card.reduced_message', [
```
Comment (Flanker72) [inline client-app/shared/account/components/product-card-reorder.vue:25]: Why all the keys in this file are inside `reorder_info_popup` namespace? Possible wrong namespace.

File: client-app/shared/bulk-order/components/copy-and-paste.vue [hunk 11]
```diff
@@ -1,19 +1,19 @@
...
   <section class="flex flex-col">
-    <h2 class="hidden lg:block px-5 py-2 border-b font-extrabold uppercase">Copy & Paste</h2>
+    <h2
+      class="hidden lg:block px-5 py-2 border-b font-extrabold uppercase"
+      v-t="'shared.bulk_order.copy_n_past.title'"
+    ></h2>
 
     <div class="p-6 pb-5 md:p-5 pt-4 h-full">
-      <p class="font-bold mb-2 text-sm">Want to save time and order faster?</p>
+      <p class="font-bold mb-2 text-sm" v-t="'shared.bulk_order.copy_n_past.subtitle_message'"></p>
```
Comment (Flanker72) [inline client-app/shared/bulk-order/components/copy-and-paste.vue:9]: Missing "e" in "paste" in the whole file.

File: client-app/shared/cart/components/cart-add-info.vue [hunk 12]
```diff
@@ -74,6 +92,8 @@ const props = defineProps({
 
 const variant = computed(() => (props.lineItem.quantity === 0 ? "warn" : "success"));
 const title = computed(() =>
-  props.lineItem.quantity === 0 ? "1 Product removed from cart" : "1 Product added to cart"
+  props.lineItem.quantity === 0
+    ? t("shared.cart.cart_add_info_popup.title_1")
```
Comment (Flanker72) [inline client-app/shared/cart/components/cart-add-info.vue:96]: Uninformative title keys: please, use "title_removed" and "title_added" instead.

File: client-app/shared/catalog/components/product-card-grid.vue [hunk 13]
```diff
@@ -23,17 +23,17 @@
...
         <div v-if="product.availabilityData?.isInStock" class="flex items-center text-green-700 text-xs">
           <div class="w-1.5 h-1.5 bg-green-700 rounded mr-1"></div>
           {{ product.availabilityData.availableQuantity > 9999 ? "9999+" : product.availabilityData.availableQuantity }}
-          in stock
+          {{ $t("common.suffixes.product_count_in_stock") }}
         </div>
         <div v-else class="flex items-center text-[color:var(--color-danger)] text-xs">
           <div class="w-1.5 h-1.5 bg-[color:var(--color-danger)] rounded mr-1"></div>
-          Out of stock
+          {{ $t("common.suffixes.product_out_of_stock") }}
```
Comment (Flanker72) [inline client-app/shared/catalog/components/product-card-grid.vue:36]: Why it is a suffix if it is a complete message?

File: client-app/shared/checkout/components/product-card.vue [hunk 14]
```diff
@@ -57,11 +57,14 @@
...
                 <span class="text-green-700 text-xs pt-1 whitespace-nowrap"
-                  >{{ lineItem.inStockQuantity! > 9999 ? "9999+" : lineItem.inStockQuantity }} in stock</span
+                  >{{ lineItem.inStockQuantity! > 9999 ? "9999+" : lineItem.inStockQuantity }}
+                  {{ $t("common.suffixes.product_count_in_stock") }}</span
                 >
               </div>
               <div v-else class="flex items-center">
-                <span class="text-[color:var(--color-danger)] text-xs pt-1 whitespace-nowrap">Out of stock</span>
+                <span class="text-[color:var(--color-danger)] text-xs pt-1 whitespace-nowrap">{{
+                  $t("common.suffixes.product_out_of_stock")
```
Comment (Flanker72) [inline client-app/shared/checkout/components/product-card.vue:66]: This is the complete message, not suffix again

File: client-app/shared/account/components/product-card-reorder.vue (hunk 9 above)
Comment (trueboroda) [reply client-app/shared/account/components/product-card-reorder.vue:185]: Becouse it is  a child component for reorder-info popup

File: client-app/shared/account/components/product-card-reorder.vue (hunk 10 above)
Comment (trueboroda) [reply client-app/shared/account/components/product-card-reorder.vue:25]: Becouse it is a child component for reorder-info popup

**Review by Flanker72 — APPROVED** [review]

**Pull request merged**